pallets/order-book/src/lib.rs

```diff
@@ -571,87 +558,86 @@ pub mod pallet {
 		}
 
 		/// Fill an existing order, based on the provided partial buy amount.
+		/// The `amount_out` is the amount the originator of this call is
+		/// willing to buy for
 		#[pallet::call_index(7)]
 		#[pallet::weight(T::Weights::fill_order_partial())]
 		pub fn fill_order_partial(
 			origin: OriginFor<T>,
 			order_id: T::OrderIdNonce,
-			buy_amount: T::Balance,
+			amount_out: T::Balance,
 		) -> DispatchResult {
 			let account_id = ensure_signed(origin)?;
 			let order = <Orders<T>>::get(order_id)?;
 
-			Self::fulfill_order_with_amount(order, buy_amount, account_id)
+			Self::fulfill_order_with_amount(order, amount_out, account_id)
 		}
 	}
 
 	impl<T: Config> Pallet<T> {
 		pub fn fulfill_order_with_amount(
 			order: OrderOf<T>,
-			buy_amount: T::Balance,
+			amount_out: T::Balance,
 			account_id: T::AccountId,
 		) -> DispatchResult {
 			ensure!(
-				buy_amount >= order.min_fulfillment_amount,
+				amount_out >= order.min_fulfillment_amount_out,
 				Error::<T>::InsufficientOrderSize,
 			);
 
-			ensure!(
-				T::TradeableAsset::can_hold(order.asset_in_id, &(), &account_id, buy_amount),
-				Error::<T>::InsufficientAssetFunds,
-			);
+			let price = match order.price {
+				OrderPrice::Market => Self::market_price(order.asset_out_id, order.asset_in_id)?,
+				OrderPrice::Custom(price) => price,
+			};
+
+			let amount_in =
+				Self::convert_with_ratio(order.asset_out_id, order.asset_in_id, price, amount_out)?;
 
-			let sell_amount = Self::convert_with_ratio(
-				order.asset_in_id,
-				order.asset_out_id,
-				order.max_sell_rate,
-				buy_amount,
-			)?;
 			let remaining_buy_amount = order
```

Comment:
Nit: Should be name `remaining_sell_amount`
```suggestion
			let remaining_sell_amount = order
```

Reply:
Actually, to follow the new convention I would say: `remaining_amount_out`